From e1aebc68eb031f3e91ed39a0b145589f1a4a1734 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Petr=20P=C3=ADsa=C5=99?= <ppisar@redhat.com>
Date: Fri, 3 Oct 2025 12:23:11 +0200
Subject: [PATCH] multisig: Rename dnf4-multisig(8) manual page to
 dnf-multisig(8)
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

To align with all other plugin manual pages.
Create dnf4-multisig(8) symlink for compatibility.

FILE(CREATE_LINK) is available since cmake 3.14.

Resolve: https://issues.redhat.com/browse/RHEL-117134
Signed-off-by: Petr Písař <ppisar@redhat.com>
---
 CMakeLists.txt        | 2 +-
 dnf-plugins-core.spec | 2 +-
 doc/CMakeLists.txt    | 4 ++++
 doc/conf.py           | 2 +-
 4 files changed, 7 insertions(+), 3 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index a1eea7b..86225e7 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1,5 +1,5 @@
 PROJECT (dnf-plugins-core NONE)
-CMAKE_MINIMUM_REQUIRED (VERSION 2.4)
+CMAKE_MINIMUM_REQUIRED (VERSION 3.14)
 
 if (NOT WITHOUT_LOCAL)
     set (WITHOUT_LOCAL "0")
diff --git a/dnf-plugins-core.spec b/dnf-plugins-core.spec
index cb3b1b8..ff6beea 100644
--- a/dnf-plugins-core.spec
+++ b/dnf-plugins-core.spec
@@ -40,7 +40,7 @@ License:        GPLv2+
 URL:            https://github.com/rpm-software-management/dnf-plugins-core
 Source0:        %{url}/archive/%{version}/%{name}-%{version}.tar.gz
 BuildArch:      noarch
-BuildRequires:  cmake
+BuildRequires:  cmake >= 3.14
 BuildRequires:  gettext
 # Documentation
 %if %{with python3}
diff --git a/doc/CMakeLists.txt b/doc/CMakeLists.txt
index 297506a..75e74bb 100644
--- a/doc/CMakeLists.txt
+++ b/doc/CMakeLists.txt
@@ -48,6 +48,10 @@ INSTALL(FILES ${CMAKE_CURRENT_BINARY_DIR}/dnf-migrate.8
 endif()
 
 if (${PYTHON_VERSION_MAJOR} STREQUAL "3")
+INSTALL(FILES ${CMAKE_CURRENT_BINARY_DIR}/dnf-multisig.8
+        DESTINATION share/man/man8)
+FILE(CREATE_LINK dnf-multisig.8 ${CMAKE_CURRENT_BINARY_DIR}/dnf4-multisig.8
+        SYMBOLIC)
 INSTALL(FILES ${CMAKE_CURRENT_BINARY_DIR}/dnf4-multisig.8
         DESTINATION share/man/man8)
 endif()
diff --git a/doc/conf.py b/doc/conf.py
index 2845d18..225ae5f 100644
--- a/doc/conf.py
+++ b/doc/conf.py
@@ -301,7 +301,7 @@ if sys.version_info[0] < 3:
     man_pages.append(('migrate', 'dnf-migrate', u'DNF migrate Plugin', AUTHORS, 8))
 
 if sys.version_info[0] == 3:
-    man_pages.append(('multisig', 'dnf4-multisig', u'DNF multisig Plugin', AUTHORS, 8))
+    man_pages.append(('multisig', 'dnf-multisig', u'DNF multisig Plugin', AUTHORS, 8))
 
 # If true, show URL addresses after external links.
 #man_show_urls = False
-- 
2.51.0

